proposals/0016-simd-application-fees.md

```diff
@@ -0,0 +1,438 @@
+---
+simd: '0016'
+title: Application Fees
+authors:
+  - Godmode Galactus (Mango Markets)
+  - Maximilian Schneider (Mango Markets)
+category: Standard
+type: Core, Fees
+status: Draft
+created: 2022-12-23
+feature:
+---
+
+
+## Summary
+
+This SIMD will discuss additional fees called Application Fees. These fees are
+decided and set by the dapp developer to interact with the dapp. Dapp developers
+then can decide to rebate these fees if a user interacts with the dapp as
+intended and disincentivize the users which do not. These fees will be applied
+even if the transaction eventually fails and are collected on the writable
+account. The owner of the account can do lamport transfer to recover these
+fees. So instead of fees going to the validator, these fees go to the **dapp
+developers**. It will be dapp developer's responsibility to advertise the
+required application fees to its users.
+
+Discussion for the issue: <https://github.com/solana-labs/solana/issues/21883>
+
+## Motivation
+
+Currently, there is no way for dapp developers to enforce appropriate behavior
+and the way their contracts should be used. Bots spamming on dapps make them
+unusable and dapps lose their users/clients because the UX is laggy and
+inefficient. Unsuccessful transactions deny block space to potentially valid
+transactions which reduces activity on the dapp. For dapps like mango or
+openbook increasing fees without any rebates or dynamically based other proposed
+mechanisms will punish potentially thousands of users because of a handful of
+malicious users. Giving dapp's authority more control to incentivize proper
+utilization of its contract is the primary motivation for this proposal.
+
+During network congestion, many transactions in the network cannot be processed
+because they all want to write-lock the same accounts. When a write lock on an
+account is taken by a transaction batch no other batch can use this account in
+parallel, so only transactions belonging to a single batch are processed
+correctly and all others are retried again or forwarded to the leader of the
+next slot. With all the forwarding and retrying the validator is choked by the
+transactions write-locking the same accounts and effectively processing valid
+transactions sequentially.
+
+There are multiple accounts of OpenBook (formerly serum), mango markets, etc
```

Comment:
Nits on "OpenBook (formerly serum)":
1. "openbook" has already been referenced, so if you want to include "(formerly serum)" then it belongs earlier
2. Be consistent. Is it OpenBook or openbook. Is it mango or mango markets